activerecord/test/cases/migration/rename_table_test.rb

```diff
@@ -111,6 +111,17 @@ def test_renaming_table_doesnt_attempt_to_rename_non_existent_sequences
           connection.drop_table :cats, if_exists: true
           connection.drop_table :felines, if_exists: true
         end
+
+        def test_renaming_table_with_long_name
+          table_name = "a" * (connection.max_identifier_length + 1)
+          connection.create_table table_name
```

Comment:
It's not necessary and might be personal preference but perhaps it would be reasonable to add an assertion or just a check like `flunk("wasn't truncated") unless <how can we check for truncation>` to make sure that truncation actually happens. Generally we are pretty confident because of relying on `max_identifier_length` but what worries me is that the test isn't going to fail if `max_identifier_length` accidentally changes to something like `10`, or an actual database limit becomes higher like 128 but the test still going to be green

Reply:
For postgres, we dynamically get a config (compared to hardcoded value for of 64 for other adapters) https://github.com/rails/rails/blob/77e6bfaa23fe5c1d585c1817d1e5c2722a3f7c87/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L492-L495

Wont' this be enough to detect changes?

Reply:
Sorry, "go to definition" confused me and I didn't check the method definition for `postgresql` adapter 🙈 
Nvm my comment, thanks for pointing out!

Reply:
> we dynamically get a config (compared to hardcoded value for of 64 for other adapters) 

This kinda confirm my hunch that we shouldn't automatically truncate, otherwise a migration could produce different index names in dev and prod if the postgres is configured differently.

Reply:
True, and it could also produce different table names, but it will be able to correctly determine this truncated name in both envs.